runtime/src/rpc.rs

```diff
@@ -638,6 +749,22 @@ pub trait UtilFuncs {
     async fn get_foreign_assets_metadata(&self) -> Result<Vec<(u32, AssetMetadata)>, Error>;
 
     async fn get_foreign_asset_metadata(&self, id: u32) -> Result<AssetMetadata, Error>;
+
+    #[cfg(any(
+        feature = "parachain-metadata-kintsugi-testnet",
+        feature = "parachain-metadata-interlay-testnet"
+    ))]
+    async fn get_lend_tokens(&self) -> Result<Vec<(CurrencyId, CurrencyId)>, Error>;
+
+    async fn get_decoded_storage_keys<T, U, F>(
```

Comment:
I'd probably not have this take a callback function, but rather an enum and have the logic inside this function. Not a huge deal if you want to keep the callback, but if so, please make it return a `Result`, since `&raw_key[BLAKE2_128_HASH_PREFIX_LENGTH..]` will panic if the input slice is shorter than expected

Reply:
You mean an enum with the hasher type of the storage key?

Reply:
yes